pyiron_base/project/archiving/export_archive.py

```diff
@@ -4,26 +4,48 @@
 
 import numpy as np
 
-from pyiron_base.project.archiving.shared import getdir
-
 
 def new_job_id(job_id, job_translate_dict):
+    """
+    Translate a job ID using a provided dictionary.
+
+    Args:
+        job_id (float or int): The job ID to be translated. If it is a float, it will be converted to an integer.
+        job_translate_dict (dict): Dictionary mapping original job IDs to new job IDs.
+
+    Returns:
+        int or None: The translated job ID if it exists in the dictionary, otherwise None.
+    """
     if isinstance(job_id, float) and not np.isnan(job_id):
         job_id = int(job_id)
-    if isinstance(job_id, int):
-        return job_translate_dict[job_id]
-    else:
-        return None
+    return job_translate_dict.get(job_id) if isinstance(job_id, int) else None
 
 
 def update_project(project_instance, directory_to_transfer, archive_directory, df):
-    directory_to_transfer = os.path.basename(directory_to_transfer)
+    """
+    Update the project paths in a DataFrame to reflect the new archive location.
+
+    Args:
+        project_instance (Project): The project instance for accessing project properties.
+        directory_to_transfer (str): The directory containing the jobs to be transferred.
+        archive_directory (str): The base directory for the archive.
+        df (DataFrame): DataFrame containing job information, including project paths.
+
+    Returns:
+        list: List of updated project paths reflecting the new archive location.
+    """
+    dir_name_transfer = os.path.basename(directory_to_transfer) or os.path.basename(
+        os.path.dirname(directory_to_transfer)
+    )
+    dir_name_archive = os.path.basename(archive_directory) or os.path.basename(
+        os.path.dirname(archive_directory)
+    )
```

Comment:
Seems that `getdir` got inlined, if it's still used in other files I'd prefer it still get used here.

Relatedly `os.path.basename(os.path.normpath(path))` seems to be a more readable definition for `getdir`.

Reply:
ok I restored it